libfwupd/fwupd-client.c

```diff
@@ -569,16 +601,10 @@ fwupd_client_curl_new (FwupdClient *self, GError **error)
 	if (g_getenv ("DISABLE_SSL_STRICT") != NULL)
 		curl_easy_setopt (helper->curl, CURLOPT_SSL_VERIFYPEER, 0L);
 
-	/* set the proxy */
-	http_proxy = g_getenv ("https_proxy");
-	if (http_proxy == NULL)
-		http_proxy = g_getenv ("HTTPS_PROXY");
-	if (http_proxy == NULL)
-		http_proxy = g_getenv ("http_proxy");
-	if (http_proxy == NULL)
-		http_proxy = g_getenv ("HTTP_PROXY");
-	if (http_proxy != NULL && strlen (http_proxy) > 0)
-		curl_easy_setopt (helper->curl, CURLOPT_PROXY, http_proxy);
+#ifdef HAVE_LIBPROXY
+	/* use libproxy to get system-wide proxy settings */
+	helper->libproxy_factory = px_proxy_factory_new ();
+#endif
```

Comment:
I think this should be an `#else` to get all the stuff from environment if you don't have libproxy though, no?  Otherwise how can Windows ever work in an environment needing proxy?

Reply:
oh nvm, it's done in the helper now, sorry for the noise.